infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/StartSynchronizer.java

```diff
@@ -287,6 +295,17 @@ private void rethrowCause(Throwable e) throws InfrastructureException {
     }
   }
 
+  /**
+   * Returns time before workspace start should be interrupted or 0 if runtime start should be
+   * interrupted now.
+   */
+  public long getStartTimeoutMillis() {
+    long currentTimeMillis = System.currentTimeMillis();
+    long deadLine = startTimeMillis + workspaceStartTimeoutMillis;
+    long timeout = deadLine - currentTimeMillis;
+    return timeout < 0 ? 0 : timeout;
```

Comment:
Code reduction proposal: `return Math.max(0, deadLine - System.currentTimeMillis());`

Reply:
> `Math.max`

+1

> Inline timeout value to return statement

I'm not sure here. I would prefer to have 3 lines instead of 2 in favor of clarity